### PR TITLE
Update margin-trim content

### LINKED ISSUE #23012: Outdated content for margin-trim
### MDN URL

https://developer.mozilla.org/en-US/docs/Web/CSS/margin-trim

### What specific section or headline is this issue about?

_No response_

### What information was incorrect, unhelpful, or incomplete?

See latest spec: https://w3c.github.io/csswg-drafts/css-box-4/#margin-trim

`all` and `in-flow` values don't exist.

### What did you expect to see?

Up-to-date content.

### Do you have any supporting links, references, or citations?

https://w3c.github.io/csswg-drafts/css-box-4/#margin-trim

### Do you have anything more you want to share?

_No response_

### MDN metadata

<details>
<summary>Page report details</summary>

* Folder: `en-us/web/css/margin-trim`
* MDN URL: https://developer.mozilla.org/en-US/docs/Web/CSS/margin-trim
* GitHub URL: https://github.com/mdn/content/blob/main/files/en-us/web/css/margin-trim/index.md
* Last commit: https://github.com/mdn/content/commit/6a4ff1ba12d84b8e369cd00fb3660bc805f93b79
* Document last modified: 2022-09-27T03:53:16.000Z

</details>

### PR DESCRIPTION
**Pull request by rohini-ranjanR**
Removing `in-flow` and `all` values issue #23012.

### Description
Removing `in-flow` and `all` values as it was not mentioned in https://w3c.github.io/csswg-drafts/css-box-4/#margin-trim

### Related issues and pull requests

Fixes #23012

### CODE REVIEW

**Comment by Josh-Cena** [comment]
We need the new property values here, like `block`, `inline`, etc. Right now there's only one value which doesn't make sense.

**Comment by rohini-ranjanR** [comment]
Can I add these in place of `in-flow` and `all`

`block`

-  Computes to 'block-start block-end'

`inline`
- Computes to 'inline-start inline-end'

**Comment by wbamberg** [comment]
> Can I add these in place of `in-flow` and `all`
> 
> `block`
> 
>     * Computes to 'block-start block-end'
> 
> 
> `inline`
> 
>     * Computes to 'inline-start inline-end'

Thank you for the update! We also need block-start, inline-start, inline-end, block-end.

You can see the alllowed values in the specification: https://w3c.github.io/csswg-drafts/css-box-4/#margin-trim.

**Comment by dipikabh** [comment]
Hello @rohini-ranjanR, thank you for your contribution.

There are a couple of changes that would need to be done to really fix https://github.com/mdn/content/issues/23012:
- Remove the outdated values
- Document the new values. Ideally, the descriptions should be more explanatory than a copy&paste from the specification. 
- Update other impacted sections such as [Examples](https://developer.mozilla.org/en-US/docs/Web/CSS/margin-trim#examples) to remove the outdated values and add code examples for the new values.

Please let us know if you would be interested in taking up this work.

**Comment by rohini-ranjanR** [comment]
Hello, @dipikabh
Yes, I would like to work on this task.

**Comment by dipikabh** [comment]
That sounds great, @rohini-ranjanR! You can push your changes/commits to this PR for this work.

Let me know if you have any questions or need any help with the documentation process.

File: files/en-us/web/css/margin-trim/index.md [hunk 1]
```diff
@@ -38,14 +42,25 @@ margin-trim: unset;
...
 
-    It also truncates any margins collapsed with such a margin.
+- `block-start`
+  - : Margins of the first block child with container's edge will be trimmed to zero.
 
-- `all`
-  - : Trims the margins of in-flow boxes and floats whose margins coincide with the container's content edge.
+- `block-end`
+
+  - : Margins of last block child with the container's edge will be truncated to zero.
```
Comment (dipikabh) [inline files/en-us/web/css/margin-trim/index.md:54]:
```suggestion
  - : Margin of last block child with the container's edge is trimmed to zero.
```

File: files/en-us/web/css/margin-trim/index.md [hunk 2]
```diff
@@ -38,14 +42,25 @@ margin-trim: unset;
...
+
+- `inline`
+  - : Container truncate the margins provided to the inline children where they adjoin the container's edge, without affecting spacing at the beginning and end of the       row.
```
Comment (dipikabh) [inline files/en-us/web/css/margin-trim/index.md:57]:
```suggestion
  - : Margins provided to the inline children where they adjoin the container's edges are trimmed to zero without affecting the spacing at the beginning and end of the row.
```

File: files/en-us/web/css/margin-trim/index.md [hunk 3]
```diff
@@ -38,14 +42,25 @@ margin-trim: unset;
...
+
+- `inline-start`
+  - : Margins between the container's edge and the first inline child will be truncated to zero.
```
Comment (dipikabh) [inline files/en-us/web/css/margin-trim/index.md:60]:
```suggestion
  - : Margin between the container's edge and the first inline child is trimmed to zero.
```

File: files/en-us/web/css/margin-trim/index.md [hunk 4]
```diff
@@ -38,14 +42,25 @@ margin-trim: unset;
...
+
+- `inline-end`
+  - : Margin between the container's edge and the last inline child will be truncated to zero.
```
Comment (dipikabh) [inline files/en-us/web/css/margin-trim/index.md:63]:
```suggestion
  - : Margin between the container's edge and the last inline child is trimmed to zero.
```

File: files/en-us/web/css/margin-trim/index.md [hunk 5]
```diff
@@ -85,14 +101,24 @@ The problem here is that you'd end up with 20px too much spacing at the right of
...
 
 ```css
 article {
-  margin-trim: in-flow;
+  margin-trim: inline-end;
+  /* … */
+}
+```
+
+Similarly, to remove left margins with the container's edge:
```
Comment (dipikabh) [inline files/en-us/web/css/margin-trim/index.md:117]:
```suggestion
Similarly, to remove left margin with the container's edge:
```

**Review by dipikabh — CHANGES_REQUESTED** [review]
Thanks @rohini-ranjanR! I've left some edit comments.
I just realized that these new values are not yet implemented in any browser and this documentation is being updated ahead of its time.

**Comment by rohini-ranjanR** [comment]
I have a doubt about the line 'The margin-trim property allows the container to trim the margins of its children where they adjoin the container's edges.' code line 20 do they should be updated to 'The margin-trim property allows to trim the margins of its children where they adjoin the container's edges. '

**Review by dipikabh — APPROVED** [review]
Thanks for the updates, @rohini-ranjanR!

**Pull request merged**